libnodegl/glcontext_eagl.m

```
@@ -208,10 +208,17 @@ static void eagl_reset_framebuffer(struct glcontext *ctx)
 {
     struct eagl_priv *eagl = ctx->priv_data;
 
-    ngli_glDeleteFramebuffers(ctx, 1, &eagl->fbo);
-    ngli_glDeleteFramebuffers(ctx, 1, &eagl->fbo_ms);
-    ngli_glDeleteRenderbuffers(ctx, 1, &eagl->depthbuffer);
-    ngli_glDeleteRenderbuffers(ctx, 1, &eagl->colorbuffer_ms);
+    if (eagl->fbo)
+        ngli_glDeleteFramebuffers(ctx, 1, &eagl->fbo);
+
+    if (eagl->fbo_ms)
+        ngli_glDeleteFramebuffers(ctx, 1, &eagl->fbo_ms);
+
+    if (eagl->depthbuffer)
+        ngli_glDeleteRenderbuffers(ctx, 1, &eagl->depthbuffer);
+
+    if (eagl->colorbuffer_ms)
+        ngli_glDeleteRenderbuffers(ctx, 1, &eagl->colorbuffer_ms);
```

You can drop this patch, glDeleteRenderbuffers() silently ignores 0 or names that do not correspond to a valid renderbuffer.

---

There is a crash in eagl_reset_framebuffer if they didn't have been generated

---

Is there a way to reproduce it ?

---

> An alternative to this patch would be to add a field fbo_initialized to the eagl_priv structure and call eagl_reset_framebuffer() and ngli_glDeleteRenderbuffers() only if fbo_initialized is set to 1. What do you think ?

I prefer the usual if null check, it's a common pattern.

---

> An alternative to this patch would be to add a field fbo_initialized to the eagl_priv structure and call eagl_reset_framebuffer() and ngli_glDeleteRenderbuffers() only if fbo_initialized is set to 1. What do you think ?

I prefer this as well because it doesn't imply the wrong assumption that these functions do not work with a 0/null parameter, but because they lack a proper gl context.

---

Ok, `fbo_initialized` seems more explicit to me in this case, in `eagl_uninit()` we will just call `eagl_reset_framebuffer()` and `ngli_glDeleteRenderbuffers()` if frame buffers has been initialized like @mbouron suggested. I agree the crash is on the glfunctions not being loaded though. I will make the modifications, if you agree on an other variable naming by tomorrow let me know